vyos/vyos-build
# T5498: fsck during boot doesnt work, part 1

## Change Summary

Included proper fsck versions and added initramfs.conf to modify level of compression of initrd.img.

This alone wont fix the issue of fsck not running during boot but will be a step in the direction of resolving this issue.

## Types of changes


## Related Task(s)

* https://vyos.dev/T5498

## Component(s) name

build, initrd

## Proposed changes

As a first fix for the issue of fsck not running on boot this commit includes all proper versions of fsck along with the logsave binary which was missing:

File `data/live-build-config/includes.chroot/etc/initramfs-tools/hooks/10-vyos-addons`:

```
# missing fsck in initramfs
copy_exec /sbin/dosfsck
copy_exec /sbin/e2fsck
copy_exec /sbin/fsck
copy_exec /sbin/fsck.exfat
copy_exec /sbin/fsck.ext2
copy_exec /sbin/fsck.ext3
copy_exec /sbin/fsck.ext4
copy_exec /sbin/fsck.fat
copy_exec /sbin/fsck.msdos
copy_exec /sbin/fsck.vfat
copy_exec /sbin/logsave
```

Also added file `data/live-build-config/includes.chroot/etc/initramfs-tools/initramfs.conf` to modify level of compression:

```
COMPRESS=zstd
COMPRESSLEVEL=19
```

Note! Current kernel config doesnt allow for other compressions than gzip (fallback) or zstd.

That have been filed as a separate task: https://vyos.dev/T5640

* Smoketest results:

```
DEBUG - ERROR: One or more tests failed!
DEBUG - vyos@vyos:~$ echo EXITCODE:$?
DEBUG - echo EXITCODE:$?
ERROR - Smoketest failed :/
ERROR - Unknown error occured while VyOS!
Traceback (most recent call last):
  File "/vyos/T5498/vyos-build/scripts/check-qemu-install", line 521, in <module>
    raise Exception("Smoketest-failed, please look into debug output")
Exception: Smoketest-failed, please look into debug output
```

Turns out that the failure is NOT related to this commit:

```
DEBUG - Running Testcase: /usr/libexec/vyos/tests/smoke/cli/test_protocols_ospf.py
DEBUG - test_ospf_01_defaults (__main__.TestProtocolsOSPF.test_ospf_01_defaults) ... ok
DEBUG - test_ospf_02_simple (__main__.TestProtocolsOSPF.test_ospf_02_simple) ... ok
DEBUG - test_ospf_03_access_list (__main__.TestProtocolsOSPF.test_ospf_03_access_list) ... ok
DEBUG - test_ospf_04_default_originate (__main__.TestProtocolsOSPF.test_ospf_04_default_originate) ... ok
DEBUG - test_ospf_05_options (__main__.TestProtocolsOSPF.test_ospf_05_options) ... ok
DEBUG - test_ospf_06_neighbor (__main__.TestProtocolsOSPF.test_ospf_06_neighbor) ... ok
DEBUG - test_ospf_07_redistribute (__main__.TestProtocolsOSPF.test_ospf_07_redistribute) ... ok
DEBUG - test_ospf_08_virtual_link (__main__.TestProtocolsOSPF.test_ospf_08_virtual_link) ... ok
DEBUG - test_ospf_09_interface_configuration (__main__.TestProtocolsOSPF.test_ospf_09_interface_configuration) ... ok
DEBUG - test_ospf_11_interface_area (__main__.TestProtocolsOSPF.test_ospf_11_interface_area) ... ok
DEBUG - test_ospf_12_vrfs (__main__.TestProtocolsOSPF.test_ospf_12_vrfs) ... ok
DEBUG - test_ospf_13_export_list (__main__.TestProtocolsOSPF.test_ospf_13_export_list) ... FAIL
DEBUG - test_ospf_14_segment_routing_configuration (__main__.TestProtocolsOSPF.test_ospf_14_segment_routing_configuration) ... ok
DEBUG - test_ospf_15_ldp_sync (__main__.TestProtocolsOSPF.test_ospf_15_ldp_sync) ... ok
DEBUG - test_ospf_16_graceful_restart (__main__.TestProtocolsOSPF.test_ospf_16_graceful_restart) ... ok
DEBUG - 
DEBUG - ======================================================================
DEBUG - FAIL: test_ospf_13_export_list (__main__.TestProtocolsOSPF.test_ospf_13_export_list)
DEBUG - ----------------------------------------------------------------------
DEBUG - Traceback (most recent call last):
DEBUG -   File "/usr/libexec/vyos/tests/smoke/cli/test_protocols_ospf.py", line 428, in test_ospf_13_export_list
DEBUG -     self.assertIn(f'router ospf', frrconfig)
DEBUG - AssertionError: 'router ospf' not found in ''
DEBUG - 
DEBUG - ----------------------------------------------------------------------
DEBUG - Ran 15 tests in 126.221s
DEBUG - 
DEBUG - FAILED (failures=1)
```

## How to test

Make sure that the image boots.

Once properly fixed (probably after step 2) adding "fsck.mode=force fsck.repair=yes" to the "/boot/grub/grub.cfg" file should make fsck to run and fix any issues with the storage that VyOS is installed at.

## Checklist:
